File: codewars/cpp/count_characters_in_your_string.cpp
// https://www.codewars.com/kata/52efefcbcdf57161d4000091/train/cpp


#include <iostream>
#include <map>

std::map<char, unsigned> count(const std::string& string) {
    std::map<char, unsigned> counter;
    for(int i = 0; i < string.size(); ++i) {
        counter[string[i]] += 1;
    }

    std::map<char, unsigned> count;
    for(auto j : counter) {
        count.insert({j.first, j.second});
    }

    return count;
}

int main() {
    std::string string;
    std::cin >> string;
    std::map<char, unsigned> c = count(string);
    
    for(auto it = c.begin(); it != c.end(); ++it) {
        std::cout << it->first << ", " << it->second << "\n";
    }

    return 0;
}

File: codewars/js/count_characters_in_your_string.js
// https://www.codewars.com/kata/52efefcbcdf57161d4000091/train/javascript

function count(string) {
    let count = {};

    for (const i in string) {
        if (!(string[i] in count)) {
            count[string[i]] = 1;
        } else {
            count[string[i]] += 1;
        }
    }

    return count;
}

console.log(count("Bee"))
